src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -329,7 +329,10 @@ void Run (DirectoryAssemblyResolver res, bool useMarshalMethods)
 					Log.LogCodedError ("XA4215", Properties.Resources.XA4215_Details, kvp.Key, typeName);
 			}
 
-			// NOTE: $(SupportedOSPlatformVersion) will potentially be 21.0
+			// NOTE: $(SupportedOSPlatformVersion) will potentially be 21.0, but could also be an int
+			if (SupportedOSPlatformVersion.IndexOf (".", StringComparison.OrdinalIgnoreCase) == -1) {
```

Comment:
It probably doesn't matter, but this is slightly faster:

```suggestion
			if (SupportedOSPlatformVersion.IndexOf ('.') == -1) {
```